plugins/autostandby.koplugin/main.lua

```diff
@@ -0,0 +1,166 @@
+local Device = require("device")
+
+if not Device:isPocketBook() --[[and not Device:isKobo()]] then
+    return { disabled = true, }
+end
+
+local PowerD = Device:getPowerDevice()
+local DataStorage = require("datastorage")
+local LuaSettings = require("luasettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local SpinWidget = require("ui/widget/spinwidget")
+local logger = require("logger")
+local _ = require("gettext")
+
+local AutoStandby = WidgetContainer:new{
+    is_doc_only = false,
+    name = "autostandby",
+    settings = LuaSettings:open(DataStorage:getSettingsDir() .. "/autostandby.lua"),
+    delay = 0,
+    lastInput = 0,
+    preventing = false,
+}
+
+function AutoStandby:init()
+    if not self.settings:has('filter') then
+        logger.dbg("AutoStandby: No settings found, initializing defaults")
+        self.settings
+        -- If forbidden, standby is never allowed to occur
+        :saveSetting('forbidden', false)
+        -- Consider input only further than this many seconds apart
+        :saveSetting('filter', 1)
+        -- Initial delay period during which we won't standby
+        :saveSetting('min', 1)
+        -- Multiply the delay with each subsequent input that happens, scales up to max
+        :saveSetting('mul', 1.5)
+        -- The multiplier is applied only if event arrives in (current_delay + win) period.
+        :saveSetting('win', 5)
+        -- Input that happens further than 30 seconds since last input one reset delay back to 'min'
+        :saveSetting('max', 30)
+        -- If battery is below this percent, make auto-standby aggressive again (disables scaling by mul)
+        :saveSetting('bat', 60)
+        :flush()
+    end
+
+    UIManager.event_hook:registerWidget("InputEvent", self)
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function AutoStandby:addToMainMenu(menu_items)
+    menu_items.autostandby = {
+        text = _("Auto-standby settings"),
+        sub_item_table = {
+            {
+                keep_menu_open = true,
+                text = _("Allow auto-standby"),
+                checked_func = function() return self:isAllowedByConfig() end,
+                callback = function() self.settings:saveSetting('forbidden', self:isAllowedByConfig()):flush() end,
+            },
+            self:_config_item(_("Min input idle seconds"), "min", function() return 0 end, function() return self.settings:readSetting("max") end),
+            self:_config_item(_("Max input idle seconds"), "max", function() return 0 end),
+            self:_config_item(_("Input window seconds"), "win", function() return 0 end, function() return self.settings:readSetting("max") end),
+            self:_config_item(_("Always standby if battery below"), "bat", function() return 0 end, function() return 100 end),
```

Comment:
In the rest of the code, such fonction is usually named getXXXMenuItem() or genXXXMenu():
```
Translator:genSettingsMenu()
ReaderHighlight:genHighlightDrawerMenu()
ReaderTypeset:genStyleSheetMenu()
sub_item_table = self:genStyleSheetMenu(),
ReaderView:genOverlapStyleMenu(overlap_enabled_func)
self:genDictionariesMenu(),
self:_genDownloadDictionariesMenu()
ReaderDictionary:_genDownloadDictionariesMenu()
ReaderDictionary:genDictionariesMenu()
genAutoSaveMenuItem(value)
genAutoSaveMenuItem(false),
genAutoSaveMenuItem(5),
genAutoSaveMenuItem(15),
genAutoSaveMenuItem(60),
```
So may be stick to that kind of name, so we know immediately what this is about.

Reply:
Agreed, something like `genConfigMenuItem` (?) would be more in line with our idiom.

Reply:
went with genSpinMenuItem (as the widget template is specifically only for spinner, other widgets would get their own generator)